cmd/swarm/download.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/cmd/utils"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/swarm/api"
+	swarm "github.com/ethereum/go-ethereum/swarm/api/client"
+	"gopkg.in/urfave/cli.v1"
+)
+
+func download(ctx *cli.Context) {
+	log.Debug("swarm down")
+	args := ctx.Args()
+	if len(args) < 1 {
+		utils.Fatalf("Usage: swarm down <bzz locator> [<destination path>]")
+	}
+
+	var (
+		bzzapi      = strings.TrimRight(ctx.GlobalString(SwarmApiFlag.Name), "/")
+		isRecursive = ctx.GlobalBool(SwarmRecursiveUploadFlag.Name)
+		client      = swarm.NewClient(bzzapi)
+	)
+
+	dir := ""
+
+	if len(args) == 1 {
+		// no destination arg - assume current terminal working dir
+		workingDir, err := filepath.Abs("./")
```

Comment:
Just initialise `dir := "."` above, there is no need to get an absolute path.

Reply:
why not use an absolute path?

Reply:
Because it is unnecessary, the OS knows that `"."` refers to the current directory, you don't need to explicitly turn it into an absolute path.